### DISCUSSION TITLE
Example integration with OpenAPI

### DISCUSSION THREAD

**Discussion by Shaun-Hirst** [keycloak/keycloak#21500] [Q&A]
I have a example openapi project on github

[https://github.com/Shaun-Hirst/OpenAPI-Example](url)

the main example currently is python flask i'm looking to add security and user permission using keycloak.

are there any examples of setting up keycloak for this or examples of using this it in python

look to build as many much as I can into this. to have a one stop shop of good practices on how to do apis. any help welcome.

**Comment by christian-2** [comment]
Have you considered [Flask-OIDC](https://flask-oidc.readthedocs.io/en/latest/), say? You'll basically need an OIDC SP (or SAML RP) on your side.

**Reply by Shaun-Hirst** [reply]
Yes I look at openid,

just looking to create something that anyone can pull and deploy without really having to do anything thing special

just a simple read and write permissions will do. I'm under the impression that keycloak can provide user auth without the need for an extra layer.

also be good to expose the adding and editing of users to the api.
